test/validators.js

```diff
@@ -8025,4 +8025,34 @@ describe('Validators', () => {
       ],
     });
   });
+
+  it('should validate date', () => {
+    test({
+      validator: 'isDate',
+      valid: [
+        new Date(),
+        new Date([2014, 2, 15]),
+        new Date('2014-03-15'),
+        '2002-07-15',
+        '2002/07/15',
+        '07/15/2002',
+        '07-15-2002',
+        '2015-07-15T07:00:00+0000',
+      ],
+      invalid: [
+        '',
```

Comment:
Try add the following test cases:
```
2020-02-30, // invalid date
2019-02-29, // non-leap year
2020-04-31, // invalid date
```

I presume all those will pass as true (or fail as false). We might to handle these cases.

Reply:
Since:
```
> new Date('2020-02-31')
2020-03-02T00:00:00.000Z
> new Date('2020-04-31')
2020-05-01T00:00:00.000Z
```

Reply:
nice catch!

a simple validation would be 

```javascript
> new Date('2020-02-30').getUTCDate()
1
> new Date('2019-02-29').getUTCDate()
1
> new Date('2020-02-29').getUTCDate()
29
```

we can compare it with the date supplied 

the problem here is with the date format some may give as YYYY-MM-DD or MM-DD-YYYY

we need to find out which one is the date

any suggestion?

Reply:
I'd not thought of that way, I think I like it! 👍

Reply:
hmm looks like there are many edge cases to be considered in this case, 

what if a user supplies

**MM-DD-YYYY
YYYY-MM-DD
YY-MM-DD
MM-DD-YY**


we need to keep things simple and general


any suggestion?

Reply:
Sure, for now let's just support those that are supported by `Date()`, and strictly YYYY. Need to add that note on the README.

However, I noticed something else, looks at this?
```
> new Date('02-02-2019')
2019-02-01T21:00:00.000Z // .getUTCDate() = 1
> new Date('2019-02-02')
2019-02-02T00:00:00.000Z // .getUTCDate() = 2
```

Perhaps this are some of the intricacies that led to deprecating the first `isDate()` from the library. 🤔

What if we allowed the user to supply the format from a list of predefined formats?
```
MM-DD-YYYY
DD-MM-YYYY
YYYY-MM-DD
```
Then transform this to a `YYYY-MM-DD` format before running it through `new Date()`

Reply:
check out the fiddle

https://jsfiddle.net/x2z3htL1/1/

I have done some tweaking, I have some stuff to do so right now I don't have time to think about edge cases 

feel free to introduce some edge case or enhance the code

let me know if you are aware of any edge cases

Reply:
Looking forward to that we can do more addition on edge cases in the future and update the readme accordingly.